# Scraping Filpkart Details Of Mobile Phones In India

![img](https://imgur.com/tEfdMsM.png)

## Introduction 

This project is about scraping Filpkart Details Of Mobile Phones In India. I have used a database from Filpkart Website and Searched Mobile in the search section for the project. The website name is https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page This website contains all the information about various phones and their details such as Product Name, Price, Ratings, Description, Image etc. I have attempted to collect the information about mobiles listed in the site (which shown in search section).

### I have Scraped And Collected the Following Information About Each Individual Mobilephone

1. Name Of The Product
2. Rating Of The Product
3. Price Of The Product
4. Image Of The Product
5. Description Of The Product

### These are The Steps that I have followed for the execution of this Project

1. Importing the necessary libraries
2. Check whether multiple pages can be accessed by the URL definition or not
3. Use of a simple while loop to get the required information
4. Using pandas create the date frame
5. Check and convert the data frame to a CSV file
6. Submit the project

## 1: Importing the necessary libraries

Here I have used Beautiful Soup Library to scrape the pages. The Beautiful Soup is a python library which is named after a Lewis Carroll poem of the same name in “Alice’s Adventures in the Wonderland”. Beautiful Soup is a python package and as the name suggests, parses the unwanted data and helps to organize and format the messy web data by fixing bad HTML and present to us in an easily-traversible XML structures. In short, Beautiful Soup is a python package which allows us to pull data out of HTML and XML documents.

I have used requests to get the url response from the site https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page 

Pandas is required to change the raw information in a clean data frame to view the CSV File

In [1]:
from bs4 import BeautifulSoup as bs
import requests
import pandas as pd

## 2: Check Wheather Multiple Pages Can Be Accessed By The URL Definition Or Not

Today I am trying to scrape a multiple pages, so I wanted to make sure all the pages can be accessed or not. In order to do that I just simply used a while with page numbers as a variable and printed out all the links

In [2]:
page = 1
while page != 10:
    url = f"https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page={page}"
    print(url)
    page = page + 1

https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page=1
https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page=2
https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page=3
https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&

## 3: Use Of Simple While loop To Get The Required Information

Here First used while to get the information about a single value from 10 pages which is series name of the solar panel. To do this I declared a list, using find_all() function I  brought the information by its class name. While loop is used to change the page number. Each time the loop executes, The information from the li tag is stored inside the list series name

![img](https://imgur.com/I8vnuU3.png)

### gerneral_list() function

The general_list() function is a generalized function which can be used for get information from multiple pages of different tags with class names. 

The parameters include, total number of pages to be scraped or the last page number, A list which can carry the collected information, the tag name from which the data has to be collected, and lastly the class name for proper collection of similar tags.

In order to go from the starting page to the last page I have used a while loop with page to be as the itiration variable which is increamented at the end. 

The Function starts with url set to the specific address, then responses are brought from this site address using requests module with the help of get() function. The whole response is set parsed as lxml using beautifulSoup module. 

Using simple find_all method, information is brought from the specific tags by using their class names and appended into the list. This list can be later saved as a dataframe which is explained below

In [3]:
def general_list(list_item_name, tag_name,class_name):
    page = 1
    list_item_name = []
    while page != 50:
        url = f"https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page={page}"
        response = requests.get(url)
        html = response.content
        soup = bs(html, "lxml")
        
        for i in soup.find_all(tag_name,{'class':class_name}):
            list_item_name.append(i.get_text(strip=True))
        page = page + 1
    return list_item_name


In [4]:
Product_name=[]
product_name=[]
product_name=general_list(Product_name,'div','_4rR01T')

I repeated the procedure to get name of the Mobile Phone information from the site. Later on I appended all the lists to another List called Mobile_details.

![img](https://imgur.com/53kkwG0.png)

In [5]:
Ratings=[]
Mobile_Ratings=[]
Mobile_Ratings=general_list(Ratings,'div','_3LWZlK')

![img](https://imgur.com/6AHqckl.png)

In [6]:
product_price=[]
Product_Price=[]
Product_Price=general_list(product_price,'div','_30jeq3 _1_WHN1')

![img](https://imgur.com/sLQW8QY.png)

In [7]:
image = []
Product_Image=[]
Product_Image=general_list(image ,'img','_396cs4')

![img](https://imgur.com/3rc5jIv.png)

In [8]:
product_description = []
Product_Description=[]
Product_Description=general_list(product_description ,'ul','_1xgFaf')

![img](https://imgur.com/XwKeqgu.png)

### Combining Different Lists Into a Single Dictionary

In [9]:
Mobile_details={"product_name":product_name,"Mobile_Ratings":Mobile_Ratings,"Product_Price":Product_Price,"Product_Image":Product_Image,"Product_Details":Product_Description}

In [10]:
Mobile_details

{'product_name': ['SAMSUNG Galaxy F04 (Jade Purple, 64 GB)',
  'POCO C50 (Royal Blue, 32 GB)',
  'POCO C50 (Country Green, 32 GB)',
  'Cellecor E6+',
  'MOTOROLA e13 (Aurora Green, 64 GB)',
  'MOTOROLA g52 (Charcoal Grey, 64 GB)',
  'MOTOROLA g52 (Metallic White, 128 GB)',
  'Cellecor E8 Plus',
  'Nokia 105 SS',
  'Infinix SMART 7 (Emerald Green, 64 GB)',
  'Infinix SMART 7 (Azure Blue, 64 GB)',
  'realme C55 (Rainy Night, 64 GB)',
  'Infinix SMART 7 (Night Black, 64 GB)',
  'MOTOROLA g31 (Meteorite Grey, 64 GB)',
  'REDMI 10 (Pacific Blue, 64 GB)',
  'Cellecor E10',
  'POCO C55 (Cool Blue, 128 GB)',
  'POCO C55 (Power Black, 128 GB)',
  'MOTOROLA e40 (Carbon Gray, 64 GB)',
  'Cellecor E2+',
  'REDMI 10 (Shadow Black, 64 GB)',
  'REDMI 10 (Midnight Black, 64 GB)',
  'POCO M4 Pro 5G (Cool Blue, 128 GB)',
  'Cellecor E9',
  'MOTOROLA g22 (Cosmic Black, 64 GB)',
  'MOTOROLA e13 (Cosmic Black, 64 GB)',
  'POCO M4 Pro 5G (Yellow, 128 GB)',
  'Cellecor E6+',
  'Nokia 105 SS',
  'MOTOROLA g73

## 4: Using Pandas To Create The Dateframe

By using pd.DataFrame() method, I created a table by passing the list Mobile_details and cheked it by printing and using head()

In [11]:
print(len(product_name))
print(len(Mobile_Ratings))
print(len(Product_Price))
print(len(Product_Image))
print(len(Product_Description))

984
1561
984
1189
984


In [12]:
df=pd.DataFrame(data=[product_name,Mobile_Ratings,Product_Price, Product_Description])
print(df.head())

                                                0     \
0            SAMSUNG Galaxy F04 (Jade Purple, 64 GB)   
1                                                4.5   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                1     \
0                       POCO C50 (Royal Blue, 32 GB)   
1                                                4.4   
2                                             ₹7,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                2     \
0                    POCO C50 (Country Green, 32 GB)   
1                                                4.4   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                3     \
0                                       Cellecor E6+   
1                                            

In [13]:
df.head()

0     \
0            SAMSUNG Galaxy F04 (Jade Purple, 64 GB)   
1                                                4.5   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                1     \
0                       POCO C50 (Royal Blue, 32 GB)   
1                                                4.4   
2                                             ₹7,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                2     \
0                    POCO C50 (Country Green, 32 GB)   
1                                                4.4   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                3     \
0                                       Cellecor E6+   
1                                                3.8   
2                                             ₹1,326   
3  32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...   

                                                4     \
0                 MOTOROLA e13 (Aurora Green, 64 GB)   
1                                                4.1   
2                                             ₹8,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                5     \
0                MOTOROLA g52 (Charcoal Grey, 64 GB)   
1                                                4.2   
2                                             ₹7,999   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                6     \
0              MOTOROLA g52 (Metallic White, 128 GB)   
1                                                4.2   
2                                            ₹10,999   
3  4 GB RAM | 64 GB ROM16.76 cm (6.6 inch) Full H...   

                                                7     \
0                                   Cellecor E8 Plus   
1                                                3.3   
2                                             ₹1,046   
3  32 MB RAM | 32 MB ROM4.5 cm (1.77 inch) Displa...   

                                                8     \
0                                       Nokia 105 SS   
1                                                4.2   
2                                             ₹7,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 2 TB16....   

                                                9     ...  1551  1552  1553  \
0             Infinix SMART 7 (Emerald Green, 64 GB)  ...  None  None  None   
1                                                4.2  ...     5   4.7     5   
2                                             ₹7,499  ...  None  None  None   
3  4 GB RAM | 64 GB ROM | Expandable Upto 2 TB16....  ...  None  None  None   

   1554  1555  1556  1557  1558  1559  1560  
0  None  None  None  None  None  None  None  
1     5   4.3     1     1   4.2     5     5  
2  None  None  None  None  None  None  None  
3  None  None  None  None  None  None  None  

[4 rows x 1561 columns]

In [14]:
df

0     \
0            SAMSUNG Galaxy F04 (Jade Purple, 64 GB)   
1                                                4.5   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                1     \
0                       POCO C50 (Royal Blue, 32 GB)   
1                                                4.4   
2                                             ₹7,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                2     \
0                    POCO C50 (Country Green, 32 GB)   
1                                                4.4   
2                                             ₹5,749   
3  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...   

                                                3     \
0                                       Cellecor E6+   
1                                                3.8   
2                                             ₹1,326   
3  32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...   

                                                4     \
0                 MOTOROLA e13 (Aurora Green, 64 GB)   
1                                                4.1   
2                                             ₹8,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                5     \
0                MOTOROLA g52 (Charcoal Grey, 64 GB)   
1                                                4.2   
2                                             ₹7,999   
3  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....   

                                                6     \
0              MOTOROLA g52 (Metallic White, 128 GB)   
1                                                4.2   
2                                            ₹10,999   
3  4 GB RAM | 64 GB ROM16.76 cm (6.6 inch) Full H...   

                                                7     \
0                                   Cellecor E8 Plus   
1                                                3.3   
2                                             ₹1,046   
3  32 MB RAM | 32 MB ROM4.5 cm (1.77 inch) Displa...   

                                                8     \
0                                       Nokia 105 SS   
1                                                4.2   
2                                             ₹7,499   
3  4 GB RAM | 64 GB ROM | Expandable Upto 2 TB16....   

                                                9     ...  1551  1552  1553  \
0             Infinix SMART 7 (Emerald Green, 64 GB)  ...  None  None  None   
1                                                4.2  ...     5   4.7     5   
2                                             ₹7,499  ...  None  None  None   
3  4 GB RAM | 64 GB ROM | Expandable Upto 2 TB16....  ...  None  None  None   

   1554  1555  1556  1557  1558  1559  1560  
0  None  None  None  None  None  None  None  
1     5   4.3     1     1   4.2     5     5  
2  None  None  None  None  None  None  None  
3  None  None  None  None  None  None  None  

[4 rows x 1561 columns]

In [15]:
tdf = df.T

In [16]:
tdf.head()

0    1       2  \
0  SAMSUNG Galaxy F04 (Jade Purple, 64 GB)  4.5  ₹5,749   
1             POCO C50 (Royal Blue, 32 GB)  4.4  ₹7,499   
2          POCO C50 (Country Green, 32 GB)  4.4  ₹5,749   
3                             Cellecor E6+  3.8  ₹1,326   
4       MOTOROLA e13 (Aurora Green, 64 GB)  4.1  ₹8,499   

                                                   3  
0  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
1  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
2  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
3  32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...  
4  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....

In [17]:
tdf.rename(columns={0:"Product_name",1:"Mobile_Ratings",2:"Product_Price",3:"Product_Description"},inplace=True)

In [18]:
tdf.shape

(1561, 4)

In [19]:
tdf

Product_name Mobile_Ratings Product_Price  \
0     SAMSUNG Galaxy F04 (Jade Purple, 64 GB)            4.5        ₹5,749   
1                POCO C50 (Royal Blue, 32 GB)            4.4        ₹7,499   
2             POCO C50 (Country Green, 32 GB)            4.4        ₹5,749   
3                                Cellecor E6+            3.8        ₹1,326   
4          MOTOROLA e13 (Aurora Green, 64 GB)            4.1        ₹8,499   
...                                       ...            ...           ...   
1556                                     None              1          None   
1557                                     None              1          None   
1558                                     None            4.2          None   
1559                                     None              5          None   
1560                                     None              5          None   

                                    Product_Description  
0     2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
1     4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
2     2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
3     32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...  
4     4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
...                                                 ...  
1556                                               None  
1557                                               None  
1558                                               None  
1559                                               None  
1560                                               None  

[1561 rows x 4 columns]

In [20]:
tdf.Product_name.fillna(tdf.Product_name.mode()[0],inplace=True)
tdf.Product_Price.fillna(tdf.Product_Price.mode()[0],inplace=True)
tdf.Product_Description.fillna(tdf.Product_Description.mode()[0],inplace=True)
tdf

Product_name Mobile_Ratings Product_Price  \
0     SAMSUNG Galaxy F04 (Jade Purple, 64 GB)            4.5        ₹5,749   
1                POCO C50 (Royal Blue, 32 GB)            4.4        ₹7,499   
2             POCO C50 (Country Green, 32 GB)            4.4        ₹5,749   
3                                Cellecor E6+            3.8        ₹1,326   
4          MOTOROLA e13 (Aurora Green, 64 GB)            4.1        ₹8,499   
...                                       ...            ...           ...   
1556                             Cellecor R3+              1        ₹1,449   
1557                             Cellecor R3+              1        ₹1,449   
1558                             Cellecor R3+            4.2        ₹1,449   
1559                             Cellecor R3+              5        ₹1,449   
1560                             Cellecor R3+              5        ₹1,449   

                                    Product_Description  
0     2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
1     4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
2     2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
3     32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...  
4     4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
...                                                 ...  
1556  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1557  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1558  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1559  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1560  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  

[1561 rows x 4 columns]

The DataFrame Contains 1561 rows and 4 columns

## 5: Check And Convert The Dataframe To CSV File

Since the data frame is clearly vissible in the above cell, Now Im converting the dataFrame to csv using to_csv() method

In [21]:
tdf.to_csv('Mobile_Phone_details.csv')

In [22]:
Mobile_df=pd.read_csv('Mobile_Phone_details.csv')

In [23]:
Mobile_df

Unnamed: 0                             Product_name  Mobile_Ratings  \
0              0  SAMSUNG Galaxy F04 (Jade Purple, 64 GB)             4.5   
1              1             POCO C50 (Royal Blue, 32 GB)             4.4   
2              2          POCO C50 (Country Green, 32 GB)             4.4   
3              3                             Cellecor E6+             3.8   
4              4       MOTOROLA e13 (Aurora Green, 64 GB)             4.1   
...          ...                                      ...             ...   
1556        1556                             Cellecor R3+             1.0   
1557        1557                             Cellecor R3+             1.0   
1558        1558                             Cellecor R3+             4.2   
1559        1559                             Cellecor R3+             5.0   
1560        1560                             Cellecor R3+             5.0   

     Product_Price                                Product_Description  
0           ₹5,749  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
1           ₹7,499  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
2           ₹5,749  2 GB RAM | 32 GB ROM | Expandable Upto 512 GB1...  
3           ₹1,326  32 MB RAM | 32 MB ROM4.57 cm (1.8 inch) Displa...  
4           ₹8,499  4 GB RAM | 64 GB ROM | Expandable Upto 1 TB16....  
...            ...                                                ...  
1556        ₹1,449  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1557        ₹1,449  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1558        ₹1,449  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1559        ₹1,449  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  
1560        ₹1,449  32 MB RAM | 32 MB ROM | Expandable Upto 32 GB6...  

[1561 rows x 5 columns]

## References, Conclusion and Future work  

### References

[1] Python official documentation. https://docs.python.org/3/

[2] Requests library. https://pypi.org/project/requests/

[3] Beautiful Soup documentation. https://www.crummy.com/software/BeautifulSoup/bs4/doc/

[4] Aakash N S, Introduction to Web Scraping, 2021. https://jovian.ai/aakashns/python-web-scraping-and-rest-api

[5] Pandas library documentation. https://pandas.pydata.org/docs/

[6] Filpkart Website. https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page

[7] Web Scraping Article. https://www.toptal.com/python/web-scraping-with-python

[8] Web Scraping Image. https://morioh.com/p/431153538ecb

[8] Working with Jupyter Notebook https://towardsdatascience.com/write-markdown-latex-in-the-jupyter-notebook-10985edb91fd

## Conclusion and Future work  

Here I have used https://www.flipkart.com/search?q=mobiles&sid=tyy%2C4io&as=on&as-show=on&otracker=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&otracker1=AS_QueryStore_OrganicAutoSuggest_1_1_na_na_na&as-pos=1&as-type=RECENT&suggestionId=mobiles%7CMobiles&requestId=ba2601a6-8ac4-4495-ba5d-e29b7af6e361&as-searchtext=M&page to collect the information about different Mobile Phone details which includes details such as Product Name, Price, Ratings, Description, Image etc. The Filkart Website also contains various other many products. In future I will try to scrape other various product details which is availble in the Filkart Website.

## 6: Submit The Project

In [25]:
import jovian

In [26]:
jovian.commit()

<IPython.core.display.Javascript object>

[jovian] Committed successfully! https://jovian.com/prajwalkarmarkar/scraping-filpkart-details-of-mobile-phones-in-india-ff72b


'https://jovian.com/prajwalkarmarkar/scraping-filpkart-details-of-mobile-phones-in-india-ff72b'